src/main/scala/org/constellation/primitives/Genesis.scala

```diff
@@ -2,81 +2,66 @@ package org.constellation.primitives
+    TypedEdgeHash(Coinbase, EdgeHashType.CheckpointHash),
+    TypedEdgeHash(Coinbase, EdgeHashType.CheckpointHash)
   )
 
-  def createGenesisTransaction(keyPair: KeyPair, allocAccountBalances: Seq[AccountBalance])(
-    implicit dao: DAO
-  ): Seq[Transaction] =
-    (allocAccountBalances.map(
-      t => dao.transactionService.createTransaction(DebtKey.address, t.accountHash, t.balance, DebtKey)
-    ) :+
-      dao.transactionService
-        .createTransaction(DebtKey.address, keyPair.getPublic.toId.address, 4e9.toLong, keyPair)).toList.sequence.unsafeRunSync
-
-  def createGenesisBlock(keyPair: KeyPair, allocAccountBalances: Seq[AccountBalance])(
-    implicit dao: DAO
-  ): CheckpointBlock =
-    CheckpointBlock.createCheckpointBlock(createGenesisTransaction(keyPair, allocAccountBalances), GenesisTips)(keyPair)
-
-  def start()(implicit dao: DAO): Unit = {
-    // TODO: Remove initial distribution
-    val fakeIdToGenerateTips = KeyUtils.makeKeyPair().getPublic.toId
-    val go = createGenesisAndInitialDistributionDirect(
-      dao.selfAddressStr,
-      Set(fakeIdToGenerateTips),
-      dao.keyPair,
-      dao.nodeConfig.allocAccountBalances
-    )
-    acceptGenesis(go, setAsTips = true)
-  }
-
-  // TODO: Get rid of this, need to add edge case for handling tips at beginning to avoid this
-  def createDistribution(
-    selfAddressStr: String,
-    ids: Seq[Id],
-    genesisSOE: SignedObservationEdge,
-    keyPair: KeyPair
-  )(implicit dao: DAO): CheckpointBlock = {
+  def createDistributionTransactions[F[_]: Sync](
+    allocAccountBalances: Seq[AccountBalance]
+  )(implicit dao: DAO): F[Seq[Transaction]] =
+    allocAccountBalances.toList
+      .traverse(ab => dao.transactionService.createTransaction(Coinbase, ab.accountHash, ab.balance, CoinbaseKey))
+      .asInstanceOf[F[Seq[Transaction]]]
 
-    val distr = ids.toList.traverse { id =>
-      dao.transactionService.createTransaction(selfAddressStr, id.address, 1e6.toLong, keyPair)
-    }.unsafeRunSync
+  def createGenesisBlock(transactions: Seq[Transaction]): CheckpointBlock =
+    CheckpointBlock.createCheckpointBlock(transactions, GenesisTips)(CoinbaseKey)
 
+  def createDistributionBlock(genesisSOE: SignedObservationEdge): CheckpointBlock =
     CheckpointBlock.createCheckpointBlock(
-      distr,
+      Seq.empty,
       Seq(
         TypedEdgeHash(genesisSOE.hash, EdgeHashType.CheckpointHash, Some(genesisSOE.baseHash)),
         TypedEdgeHash(genesisSOE.hash, EdgeHashType.CheckpointHash, Some(genesisSOE.baseHash))
       )
-    )(keyPair)
+    )(CoinbaseKey)
 
+  def createGenesisObservation(
+    allocAccountBalances: Seq[AccountBalance] = Seq.empty
+  )(implicit dao: DAO): GenesisObservation = {
+    // TODO: Get rid of unsafeRunSync
+    val txs = createDistributionTransactions[IO](allocAccountBalances).unsafeRunSync()
+    val genesis = createGenesisBlock(txs)
+    GenesisObservation(
+      genesis,
+      createDistributionBlock(genesis.soe),
```

Comment:
Do you mind changing the name to `createEmptyBlock`? It's weird to read it as `distribution block` is actually empty. wdyt?

Reply:
Totally makes sense :)